### ISSUE TITLE
Running tflite model in python directly

### ISSUE THREAD

**Issue by allan1224** [AbhayKoushik/RealTimeSleepStaging#3]
Hi, 

I'm trying to run the .tflite model directly in python, but I'm unsure how to get it running properly. I'm using tflite_runtime package on a raspberry pi. So far I was able to load the model and get its input and output details. Specifically, the input array has the shape ([   1,    1, 3000,    1]), and the output has a shape ([1, 1, 5]). I simply tried to create an array of random samples with the same input shape, but I'm getting an error: 

> RuntimeError: external/org_tensorflow/tensorflow/lite/kernels/reshape.cc:58 stretch_dim != -1 (0 != -1)Node number 42 (RESHAPE) failed to prepare.

Here's how I'm loading it:

```
interpreter = tf.lite.Interpreter(model_path="Sleep_CNN_40.tflite")
interpreter.allocate_tensors()

input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

input_shape = input_details[0]['shape']
input_data = np.array(np.random.random_sample(input_shape), dtype=np.float32)
interpreter.set_tensor(input_details[0]['index'], input_data)

interpreter.invoke()

output_data = interpreter.get_tensor(output_details[0]['index'])
print(output_data)
```

Basically I'm having trouble formatting the input to feed the model. Ultimately I want to feed it the 30-second epoch of normalized EEG data, but it's not even working on a test sample. I'm wondering why it's not working, if the input to the model takes the shape ([   1,    1, 3000,    1]) and my sample data is the same shape. I'm pretty inexperienced with tensor flow so I'm not really sure what I'm doing when it comes to reshaping array sizes and all that. 

Any feedback on how to run the tflite model directly in python would be appreciated.

Thanks

**Comment by AbhayKoushik** [comment]
Hey Allan, thanks for reaching out - given the novelty and lack of full traceback of your error presented in this issue, here are a few suggestions I have:

- Try printing the input and output arrays and see if the data seems valid.
- Track the exact line of code where the error is happening and figure out the traceback, so that will help you pinpoint the flow of errors precisely. Even with little knowledge of tensorflow you should be able to track the error. 
- Check the following resources out as they seem relevant to your error:
> [Stackoverflow](https://stackoverflow.com/questions/62386682/node-number-x-reshape-failed-to-prepare-tensor-resize-with-tflite-v2-2)
> [GitHub](https://github.com/tensorflow/tensorflow/issues/22377)

Good luck!

**Comment by allan1224** [comment]
Thanks for your suggestions, this helped me solve the problem. TFLite doesn't really support dynamic array shapes, so although the keras model was able to convert, it couldn't run properly. I installed TF-nightly where apparently it supports dynamic array shapes and that fixed the problem.  

Also just for clarification, is the output of the model the probability of the data belonging to a certain class 
[Wake,N1,N2,N3, N4] ? 

Thanks

**Comment by AbhayKoushik** [comment]
Glad it worked out. Yes, the output is more of what is called 'Confidence' of prediction expressed as percentage. Closing this issue.

Stay safe and have fun exploring!
